runtime/near-wallet-contract/implementation/wallet-contract/src/tests/emulation.rs

```diff
@@ -0,0 +1,145 @@
+    let final_other_balance = other_wallet.inner.as_account().view_account().await?.balance;
+
+    // Receiver balance increases
+    assert_eq!(
+        final_other_balance.as_yoctonear(),
+        initial_other_balance.as_yoctonear() + TRANSFER_AMOUNT.as_yoctonear()
+    );
+    // Sender balance decreases (by a little more than the
+    // `TRANSFER_AMOUNT` due to gas spent to execute the transaction)
+    let diff = NearToken::from_yoctonear(
+        initial_wallet_balance.as_yoctonear()
+            - (final_wallet_balance.as_yoctonear() + TRANSFER_AMOUNT.as_yoctonear()),
+    );
+    assert!(diff < NearToken::from_millinear(2));
+
+    Ok(())
+}
+
+// The Wallet Contract should understand the ERC-20 standard and map
+// it to NEP-141 function calls.
+#[tokio::test]
+async fn test_erc20_emulation() -> anyhow::Result<()> {
+    const MINT_AMOUNT: NearToken = NearToken::from_near(100);
+    const TRANSFER_AMOUNT: NearToken = NearToken::from_near(32);
+
+    let TestContext {
+        worker,
+        wallet_contract,
+        wallet_sk,
+        wallet_address,
+        wallet_contract_bytes,
+        ..
+    } = TestContext::new().await?;
+
+    let token_contract = nep141::Nep141::deploy(&worker).await?;
+    token_contract.mint(wallet_contract.inner.id(), MINT_AMOUNT.as_yoctonear()).await?;
+
+    // Check balance
+    let transaction = aurora_engine_transactions::eip_2930::Transaction2930 {
+        nonce: 0.into(),
+        gas_price: 0.into(),
+        gas_limit: 0.into(),
+        to: Some(Address::new(account_id_to_address(
+            &token_contract.contract.id().as_str().parse().unwrap(),
+        ))),
+        value: Wei::zero(),
+        data: [
+            crate::eth_emulation::ERC20_BALANCE_OF_SELECTOR.to_vec(),
+            ethabi::encode(&[ethabi::Token::Address(wallet_address)]),
+        ]
+        .concat(),
+        chain_id: CHAIN_ID,
+        access_list: Vec::new(),
+    };
+    let signed_transaction = crypto::sign_transaction(transaction, &wallet_sk);
+
+    let result = wallet_contract
+        .rlp_execute(token_contract.contract.id().as_str(), &signed_transaction)
+        .await?;
+
+    let balance: U128 = serde_json::from_slice(result.success_value.as_ref().unwrap())?;
+    assert_eq!(balance.0, token_contract.ft_balance_of(wallet_contract.inner.id()).await?);
+
+    // Do a transfer to another account
+    let (other_wallet, other_address) =
+        TestContext::deploy_wallet(&worker, &wallet_contract_bytes).await?;
+    token_contract.mint(other_wallet.inner.id(), MINT_AMOUNT.as_yoctonear()).await?;
+    let transaction = aurora_engine_transactions::eip_2930::Transaction2930 {
+        nonce: 1.into(),
+        gas_price: 0.into(),
+        gas_limit: 0.into(),
+        to: Some(Address::new(account_id_to_address(
+            &token_contract.contract.id().as_str().parse().unwrap(),
+        ))),
+        value: Wei::zero(),
```

Comment:
The `value` field is not yoctoNEAR, it is 10^6 * yoctoNEAR. The reason for this is because $NEAR has 24 decimal places, but base tokens in EVM chains have 18 decimals.

When constructing a transaction the `value` is set equal to `yoctoNEAR / 10^6` (integer division) and the `yoctoNEAR % 10^6` remainder is listed directly in the Near action that is being encoded by the transaction. So if less than 1 million yoctoNEAR is being used by the transaction then the value is set to zero.